tests/phpunit/tests/template.php

```diff
@@ -636,6 +655,32 @@ public function assertTemplateHierarchy( $url, array $expected, $message = '' )
 		$this->assertSame( $expected, $hierarchy, $message );
 	}
 
+	/**
+	 * @ticket 58866
+	 * @covers ::locate_template
+	 */
+	public function test_locate_template_path_for_parent_theme() {
+		$this->set_up_test_theme_root();
+		$theme = wp_get_theme( 'page-templates' );
+		switch_theme( $theme['Template'], $theme['Stylesheet'] );
+		$file_path = locate_template( 'style.css' );
+		$this->assertStringContainsString( 'themedir1/page-templates/style.css', $file_path );
+		$this->tear_down_test_theme_root();
```

Comment:
This call has to happen before any assertions, otherwise it may not be reached and cause other tests to fail.